src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivitySourceTests.cs

```diff
@@ -1073,6 +1073,147 @@ public void TestAddingMorePropertiesWithActivityCreate()
             }).Dispose();
         }
 
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void TestTraceState()
+        {
+            RemoteExecutor.Invoke(() => {
+
+                using ActivitySource aSource = new ActivitySource("SourceActivityTraceStateTest");
+                Activity.Current = null;
+
+                using ActivityListener listener = new ActivityListener();
+                listener.ShouldListenTo = (activitySource) => object.ReferenceEquals(aSource, activitySource);
+                ActivitySource.AddActivityListener(listener);
+
+                bool setTraceWithStringParent = false;
+                string? traceStateValueWithStringParent = null;
+
+                SampleActivity<string> sampleUsingParentId = (ref ActivityCreationOptions<string> activityOptions) =>
+                {
+                    if (setTraceWithStringParent)
+                    {
+                        activityOptions = activityOptions with { TraceState = traceStateValueWithStringParent };
+                    }
+                    return ActivitySamplingResult.AllDataAndRecorded;
+                };
+                listener.SampleUsingParentId = sampleUsingParentId;
+
+                bool setTraceWithContextParent = false;
+                string? traceStateValueWithContextParent = null;
+
+                SampleActivity<ActivityContext> sampleUsingContext = (ref ActivityCreationOptions<ActivityContext> activityOptions) => {
+                    if (setTraceWithContextParent)
+                    {
+                        activityOptions = activityOptions with { TraceState = traceStateValueWithContextParent };
```

Comment:
Just to make sure we are on the same page:

We've discussed about this C# 10 language feature and the understanding is: if we are compiling against an old version of runtime which doesn't provide full support for all C# 10 features, we would get compile time error if such C# 10 feature is not going to work on the target runtime. In other words, if we don't get compile time error, then we have confidence that such feature would work fine (not having runtime error) on the platforms we targeted.

Reply:
Regarding why I would call this out - there are some concerns/questions which is beyond my knowledge https://github.com/open-telemetry/opentelemetry-dotnet/discussions/2790#discussion-3800306.

Reply:
> Regarding why I would call this out - there are some concerns/questions which are beyond my knowledge https://github.com/open-telemetry/opentelemetry-dotnet/discussions/2790#discussion-3800306.

I replied there too. I will be happy to help with any questions raised there too.